site/content/entry/citra-progress-report-2018-q2.md

```diff
@@ -0,0 +1,305 @@
+
+## Open Source System Archives ([#3977](https://github.com/citra-emu/citra/pull/3977), [#3881](https://github.com/citra-emu/citra/pull/3881)) by [B3n30](https://github.com/B3n30)
+
+Many Nintendo 3DS games require the use of system archives, that contain things
+like fonts, a bad word list, and assets for Miis. Because these are copyrighted,
+Citra could not include them, and users had to dump a copy from their consoles,
+which was a generally tedious and error-prone process.
+
+Now, [B3n30](https://github.com/B3n30) has created open source alternative fonts
+and a bad word list that can be used in games instead of the official archives.
+And, because these are open source, they are now included with Citra. If a game
+requires either archive, and you haven't dumped them yet, it'll automatically give
+the game the alternative archives, further decreasing Citra's dependency on
+copyrighted material.
+
+## [Implement shadow map](https://github.com/citra-emu/citra/pull/3778) by [wwylele](https://github.com/wwylele)
+
+Shadow mapping is a way to quickly apply shadows to 3D scenes. It first renders
+the scene without any lighting, texture, or colour information, with a virtual
+camera from where the light source begins. Then, from that rendering the depth
+map is extracted. That is then applied as a texture to the darkened scene, making
+areas of it directly visible from the light source appear more brightly lit.
+
+The PICA200 and OpenGL both implement this in different ways, particularly in
+two important areas:
+
+ * The first is that PICA200 supports a variant called "soft shadows", where the
+depth map is blurred before being applied to the scene. This results in shadows
+that seem less jagged and sharp, making the light source feel more diffuse and
+evenly spread out throughout the scene. OpenGL doesn't support this at all.
+
+ * The second is that the PICA200 stores depth maps intended for shadow mapping as
+plain textures in the RGBA8 format. A lot of games exploit this in order to quickly
+convert other types of textures from an internal format to RGBA8. But, OpenGL
+stores these maps in a format internal to that graphics card, like any other map.
+
+The first point isn't as significant, since the softness can be ignored on OpenGL,
+resulting in a very fast (but inaccurate) shadow mapping. Because games rarely
+use soft shadows, this can be ignored relatively safely.
+
+The second point though, because games use it very often, it has to be implemented
+accurately. The naïve way would be to simply convert internal textures to RGBA8
+manually, but this would slow rendering down to a crawl. The smart way would be
+to try and trick the graphics card driver into performing a similar conversion,
+but this exposed a lot of bugs in it, leading to instability.
+
+After some very hard work by [wwylele](https://github.com/wwylele), both problems
+were solved by one stone. OpenGL supports an extension called Image Load/Store,
```

Comment:
`by one stone` isn't syntactically close enough to `kill two birds with one stone` here. While less pretty, `both problems were solved with a single solution`?